plugins/services/src/js/pages/task-details/TaskLogsContainer.js

```diff
@@ -50,19 +50,39 @@ class TaskLogsContainer extends mixin(StoreMixin) {
       return <Loader />;
     }
 
-    const {props} = this;
+    const {
+      directory,
+      params,
+      routes,
+      selectedLogFile,
+      task
+    } = this.props;
     const config = ConfigStore.get('config');
     const loggingStrategy = findNestedPropertyInObject(
       config,
       'uiConfiguration.plugins.mesos.logging-strategy'
     );
 
     if (loggingStrategy === SYSTEM_LOGS) {
-      return <TaskSystemLogsContainer {...props} />;
+      return <TaskSystemLogsContainer task={task} />;
     }
 
-    return <TaskFileViewer limitLogFiles={['stdout', 'stderr']} {...props} />;
+    return (
+      <TaskFileViewer
+        directory={directory}
+        limitLogFiles={['stdout', 'stderr']}
+        params={params}
+        routes={routes}
+        selectedLogFile={selectedLogFile}
+        task={task} />
+    );
   }
 }
 
+TaskFileViewer.propTypes = {
```

Comment:
We are adding propTypes to the wrong object here. Also, I think we should add all the props to the propTypes, e.g. routes, params etc.